prac_05/emails.py

```diff
@@ -0,0 +1,30 @@
+"""
+User Email Storage
+Estimate: 25 minutes
+Actual:   35 minutes
+"""
+
+def extract_name(email):
+    """Extracts name from email address."""
+    name_parts = email.split('@')[0].split('.')
+    name = ' '.join(name_parts).title()
+    return name
+
+# Initialize dictionary to store emails and names
+user_info = {}
+
+# Input loop
+email = input("Email: ")
+while email != "":
+    name_from_email = extract_name(email)
+    user_name = input(f"Is your name {name_from_email}? (Y/n) ").strip().lower()
+    if user_name == 'n':
+        user_name = input("Name: ").strip().title()
+    elif user_name != 'y':
```

Comment:
I believe there might be code missing to handle what the program does when the user enters 'Y' or when the user directly presses enter?